backend/src/main/java/com/example/cms/ticket/TicketRoleSpecification.java

```diff
@@ -0,0 +1,43 @@
+package com.example.cms.ticket;
+
+import com.example.cms.page.Page;
+import com.example.cms.security.Role;
+import lombok.AllArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@AllArgsConstructor
+public class TicketRoleSpecification implements Specification<Ticket> {
+
+    private Role role;
+    private Collection<Long> handlesPages;
+    private String email;
+
+    @Override
+    public Predicate toPredicate(Root<Ticket> root, CriteriaQuery<?> query, CriteriaBuilder criteriaBuilder) {
+        List<Predicate> predicates = new ArrayList<>();
+
+        Predicate conjunctionPred = criteriaBuilder.conjunction();
+
+        if (this.role == null) {
```

Comment:
To review later: I think that not logged in user should not be able to see tickets list, but we can confirm on our weekly call